Admin/PageAdmin.php

```diff
@@ -353,33 +353,45 @@ protected function configureFormFields(FormMapper $formMapper)
         if ($this->hasSubject() && !$this->getSubject()->isInternal()) {
             $formMapper
                 ->with('form_page.group_main_label')
-                    ->add('type', 'sonata_page_type_choice', array('required' => false))
+//                    ->add('type', 'sonata_page_type_choice', array('required' => false))
+                    ->add('type', method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix') ?
+                        'Sonata\PageBundle\Form\Type\PageTypeChoiceType' :
+                        'sonata_page_type_choice',
+                        array('required' => false))
                 ->end()
             ;
         }
 
         $formMapper
             ->with('form_page.group_main_label')
-                ->add('templateCode', 'sonata_page_template', array('required' => true))
+//                ->add('templateCode', 'sonata_page_template', array('required' => true))
+                ->add('templateCode', method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix') ?
+                    'Sonata\PageBundle\Form\Type\TemplateChoiceType' :
+                    'sonata_page_template',
+                    array('required' => true))
             ->end()
         ;
 
         if (!$this->getSubject() || ($this->getSubject() && $this->getSubject()->getParent()) || ($this->getSubject() && !$this->getSubject()->getId())) {
             $formMapper
                 ->with('form_page.group_main_label')
-                    ->add('parent', 'sonata_page_selector', array(
-                        'page' => $this->getSubject() ?: null,
-                        'site' => $this->getSubject() ? $this->getSubject()->getSite() : null,
-                        'model_manager' => $this->getModelManager(),
-                        'class' => $this->getClass(),
-                        'required' => false,
-                        'filter_choice' => array('hierarchy' => 'root'),
-                    ), array(
-                        'admin_code' => $this->getCode(),
-                        'link_parameters' => array(
-                            'siteId' => $this->getSubject() ? $this->getSubject()->getSite()->getId() : null,
-                        ),
-                    ))
+//                    ->add('parent', 'sonata_page_selector', array(
+                    ->add('parent', method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix') ?
+                        'Sonata\PageBundle\Form\Type\PageSelectorType' :
+                        'sonata_page_selector',
+                        array(
```

Comment:
This violates PSR2, and is why the diff looks so big : either put all the arguments of `add` on the same "line", or put one argument per line, but not both, please.